test/icinga-notification.cpp

```diff
@@ -102,4 +149,63 @@ BOOST_AUTO_TEST_CASE(type_filter)
 	std::cout << "#4 Notification type: " << ftype << " against " << notification->GetTypeFilter() << " must fail." << std::endl;
 	BOOST_CHECK(!(notification->GetTypeFilter() & ftype));
 }
+
+BOOST_AUTO_TEST_CASE(duplicate_dueto_filter)
+{
+	DuplicateDueToFilterHelper helper;
+
+	helper.s->SetState(ServiceCritical, true);
+	Application::GetTP().Start();
+	helper.n->BeginExecuteNotification(NotificationProblem, nullptr, false, false, "", "");
+	Application::GetTP().Stop();
+	BOOST_CHECK(helper.called == 1u);
+
+	helper.s->SetState(ServiceWarning, true);
+	Application::GetTP().Start();
+	helper.n->BeginExecuteNotification(NotificationProblem, nullptr, false, false, "", "");
+	Application::GetTP().Stop();
+	BOOST_CHECK(helper.called == 1u);
```

Comment:
The service `state` doesn't have a storage and calling its setter method will just do nothing. Adding this check brings test to fail!
```c++
BOOST_CHECK(helper.n->GetLastNotifiedStatePerUser()->Get(helper.u->GetName()) == ServiceCritical);
```